abi/src/test/java/org/web3j/abi/FunctionReturnDecoderTest.java

```diff
@@ -1298,4 +1298,40 @@ public void testDecodeTupleOfStaticArrays() {
                                 new Uint256(0),
                                 new Uint256(0))));
     }
+
+    /**
+     * Add extra test case to cover discovered edge-case issue
+     *
+     * Ref: https://github.com/web3j/web3j/issues/1858
+     * */
+    @Test
+    public void testDecodeDynamicArrayOfMatches() {
+        String rawInput =
+                "0x"
+                        + "0000000000000000000000000000000000000000000000000000000000000020"  // not sure about this parameter - dynamic array identifier?
```

Comment:
yes, this specifies the value `20` in hex which is `32` in decimal which is the offset of the start of the dynamic type